### PR TITLE
feat: start-vm: add ability to create copy-on-write ephemeral disks

### PR DESCRIPTION
**Pull request by nkraetzschmar**
**What this PR does / why we need it**:

Adds an optional parameter qcow=size to the start-vm image arguments. Providing this parameter will cause start-vm to create a copy-on-write file as an overlay of the input image and use that as a disk instead.

This has two uses:

1. Allows for testing images without modifying the build artifacts
2. Allows for simpler resizing of disk images during tests

**Definition of Done:**
- [ ] The code is sufficiently documented
- [ ] Shared the changes with the Team so everyone is aware
- [ ] The code is appropriately tested
- [ ] Checked if the code needs to be backportet to release branches of maintained versions (perform the actual backport after the merge to `main`)

### CODE REVIEW

File: bin/start-vm [hunk 1]
```diff
@@ -241,6 +241,7 @@ set_binaries () {
     bin_stat="stat"
     bin_sed="sed"
     bin_uuid="cat /proc/sys/kernel/random/uuid"
+    bin_realpath="grealpath"
```
Comment (fwilhe) [inline bin/start-vm:244]: the `g` prefix is macos/homebrew specific, on linux the binary is called `realpath`

File: bin/start-vm [hunk 2]
```diff
@@ -255,6 +256,7 @@ set_binaries () {
         bin_stat="gstat"
         bin_sed="gsed"
         bin_uuid="/usr/bin/uuidgen"
+        bin_realpath="grealpath"
```
Comment (fwilhe) [inline bin/start-vm:259]: the `g` prefix is macos/homebrew specific, on linux the binary is called `realpath`

**Comment by fwilhe** [comment]
Thanks for providing a fix @nkraetzschmar.

The syntax was a bit non-obvious to me, but it seems to work like this

```bash
./bin/start-vm .build/kvm_dev_usi-arm64-today-a9cc5c03.raw,qcow=3G
```

Seems to work fine on mac, please look into the grealpath issue and please amend the help text to make it easier to use.

File: bin/start-vm (hunk 1 above)
Comment (nkraetzschmar) [reply bin/start-vm]: good catch, yeah should be realpath for linux and grealpath on macos, having grealpath in both places was a typo

**Review by fwilhe — APPROVED** [review]
LGTM

**Review by fwilhe — CHANGES_REQUESTED** [review]
tiny thing: just noticed that we dont have qcow2 in the gitignore, leading to this git status

```
On branch feat/start-vm/qcow2
Your branch is up to date with 'origin/feat/start-vm/qcow2'.

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	bin/028f950e83f1.qcow2
```

Please add qcow2 to the gitignore file and I'll approve again

**Comment by fwilhe** [comment]
btw, why is this qcow2 file in `bin`? I see this is the pwd of the `start-vm` script and also the efivars are in there, but it seems wrong to me. would it not make sense to have it in `.build`?

**Review by fwilhe — APPROVED** [review]

**Pull request merged**